examples/java-matter-controller/java/src/com/matter/controller/commands/common/Command.java

```diff
@@ -174,16 +189,51 @@ public final int addArgument(String name, StringBuffer out, @Nullable String des
    * @param args Supplied command-line arguments as an array of String objects.
    */
   public final void initArguments(int argc, String[] args) {
-    int argsCount = mArgs.size();
+    int argvExtraArgsCount = argc;
+    int mandatoryArgsCount = 0;
+    int optionalArgsCount = 0;
+
+    for (Argument arg : mArgs) {
+      if (arg.isOptional()) {
+        optionalArgsCount++;
+      } else {
+        mandatoryArgsCount++;
+        argvExtraArgsCount--;
+      }
+    }
 
-    if (argsCount != argc) {
+    if (argc < mandatoryArgsCount) {
       throw new IllegalArgumentException(
-          "Wrong arguments number: " + argc + " instead of " + argsCount);
+          "initArguments: Wrong arguments number: " + argc + " instead of " + mandatoryArgsCount);
     }
 
-    for (int i = 0; i < argsCount; i++) {
+    // Initialize mandatory arguments
+    for (int i = 0; i < mandatoryArgsCount; i++) {
       initArgument(i, args[i]);
     }
+
+    // Initialize optional arguments
+    // Optional arguments expect a name and a value, so i is increased by 2 on every step.
+    for (int i = mandatoryArgsCount; i < argc; i += 2) {
+      for (int j = mandatoryArgsCount; j < mandatoryArgsCount + optionalArgsCount; j++) {
+        // optional arguments starts with OPTIONAL_ARGUMENT_PREFIX
+        if (args[i].length() <= OPTIONAL_ARGUMENT_PREFIX_LENGTH
```

Comment:
This condition does not depend on j, yet  it is inside the for loop of j. This seems wrong.